frame/staking-rewards/src/lib.rs

```diff
@@ -262,6 +287,18 @@ pub mod pallet {
 	pub type PendingStakers<T: Config> =
 		StorageMap<_, Twox64Concat, InstanceIdOf<T>, (), OptionQuery>;
 
+	/// amount of assets which will be added to position on next epoch
+	#[pallet::storage]
+	#[pallet::getter(fn pending_amount_extensions)]
+	pub type PendingAmountExtensions<T: Config> =
+		StorageMap<_, Twox64Concat, InstanceIdOf<T>, T::Balance, OptionQuery>;
+
+	/// time to extend position with
+	#[pallet::storage]
+	#[pallet::getter(fn pending_duration_extensions)]
+	pub type PendingDurationExtensions<T: Config> =
```

Comment:
Why do we need to track the pending duration extensions? Is it because the reward multiplier is affected by this?

Reply:
total stake is depend on this. you should not modify total stake until reward happend.